app/lib/common/utilities/pdf_utils.dart

```diff
@@ -0,0 +1,173 @@
+import 'dart:io';
+
+import 'package:flutter_share/flutter_share.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:pdf/pdf.dart';
+import 'package:pdf/widgets.dart' as pw;
+import 'package:pdf/widgets.dart';
+
+import '../module.dart';
+
+Future<String> createPdf(MedicationWithGuidelines medication) async {
+  final pdf = pw.Document();
+  pdf.addPage(
+    pw.MultiPage(
+      pageFormat: PdfPageFormat.a4,
+      build: (context) => [buildPdfPage(context, medication)],
+    ),
+  );
+  final dir = Platform.isAndroid
+      ? await getExternalStorageDirectory()
+      : await getApplicationDocumentsDirectory();
+
+  final file = File('${dir!.path}/${medication.name}.pdf');
+  await file.writeAsBytes(await pdf.save());
+  return file.path;
+}
+
+Future<void> sharePdf(MedicationWithGuidelines medication) async {
+  final path = await createPdf(medication);
+  await FlutterShare.shareFile(title: medication.name, filePath: path);
+}
+
+pw.Widget buildPdfPage(
+  pw.Context context,
+  MedicationWithGuidelines medication,
+) {
+  final relevantGuidelines = filterUserGuidelines(medication).guidelines;
+  return pw.Wrap(
+    children: [
+      _PdfSegment(
+        child: pw.Text(
+          'Personal PGX Report',
+          style: pw.TextStyle(
+            fontWeight: pw.FontWeight.bold,
+            fontSize: 30,
+          ),
+          textAlign: pw.TextAlign.center,
+        ),
+      ),
+      pw.SizedBox(height: 20, width: double.infinity),
+      _PdfSegment(
+        child: pw.Text(
+          medication.name,
+          style: pw.TextStyle(fontSize: 26),
+        ),
+      ),
+      if (medication.drugclass.isNotNullOrBlank) ...[
+        pw.SizedBox(height: 8, width: double.infinity),
+        _PdfSegment(
+          child: pw.Text(
+            medication.drugclass!,
+            style: pw.TextStyle(fontSize: 16),
+          ),
+        ),
+      ],
+      if (medication.description.isNotNullOrBlank) ...[
+        pw.SizedBox(height: 8, width: double.infinity),
+        _PdfSegment(
+          child: pw.Text(
+            medication.description!,
+            style: pw.TextStyle(fontSize: 12),
+          ),
+        )
+      ],
+      pw.SizedBox(height: 32, width: double.infinity),
+      for (final guideline in relevantGuidelines) ...[
+        ..._buildGuidelinePart(guideline),
+        pw.Divider(color: PdfColors.grey500),
```

Comment:
This can be moved into `_buildGuidelinePart` as the last element so that you don't need to do a double spread